backend/tests/unit/controllers/posts-controller.test.js

```diff
@@ -0,0 +1,424 @@
+  it('should get featured posts', async () => {
+    const req = {};
+
+    const mockFeaturedPosts = [
+      {
+        title: 'Test Post - 1',
+        authorName: 'Test Author - 1',
+        imageLink: 'https://www.forTestingPurposeOnly.my-image-1.jpg',
+        categories: [validCategories[0]],
+        description: 'This is a test post.',
+        isFeaturedPost: true,
+      },
+      {
+        title: 'Test Post - 2',
+        authorName: 'Test Author - 2',
+        imageLink: 'https://www.forTestingPurposeOnly.my-image-2.jpg',
+        categories: [validCategories[0]],
+        description: 'This is a test post.',
+        isFeaturedPost: true,
+      }
+    ];
+
+    Post.find = jest.fn().mockResolvedValueOnce(mockFeaturedPosts);
+
+    await getFeaturedPostsHandler(req, res);
+
+    expect(res.status).toHaveBeenCalledWith(200);
+    expect(res.json).toHaveBeenCalledWith(mockFeaturedPosts);
+  });
+
+  it('should handle errors', async () => {
+    const req = {};
+
+    const errorMessage = 'Internal Server Error';
+    Post.find = jest.fn().mockRejectedValueOnce(new Error(errorMessage));
+
+    await getFeaturedPostsHandler(req, res);
+
+    expect(res.status).toHaveBeenCalledWith(500);
+    expect(res.json).toHaveBeenCalledWith({ message: errorMessage });
+  });
+});
+
+describe('getPostByCategoryHandler', () => {
+  it('should get posts by category', async () => {
+    const req = { params: { category: validCategories[0] } };
+
+    const mockPosts = [
+      {
+        title: 'Test Post - 1',
+        authorName: 'Test Author - 1',
+        imageLink: 'https://www.forTestingPurposeOnly.my-image-1.jpg',
+        categories: [validCategories[0]],
+        description: 'This is a test post.',
+        isFeaturedPost: false,
+      },
+      {
+        title: 'Test Post - 2',
+        authorName: 'Test Author - 2',
+        imageLink: 'https://www.forTestingPurposeOnly.my-image-2.jpg',
+        categories: [validCategories[0]],
+        description: 'This is a test post.',
+        isFeaturedPost: false,
+      },
+      {
+        title: 'Test Post - 3',
+        authorName: 'Test Author - 3',
+        imageLink: 'https://www.forTestingPurposeOnly.my-image-3.jpg',
+        categories: [validCategories[0]],
+        description: 'This is a test post.',
+        isFeaturedPost: true,
+      }
+    ];
+
+    Post.find = jest.fn().mockResolvedValueOnce(mockPosts);
+
+    await getPostByCategoryHandler(req, res);
+
+    expect(res.status).toHaveBeenCalledWith(200);
+    expect(res.json).toHaveBeenCalledWith(mockPosts);
+  });
+
+  it('should handle errors - Invalid Category', async () => {
+    const req = { params: { category: 'Invalid Category' } };
+
+    await getPostByCategoryHandler(req, res);
+
+    expect(res.status).toHaveBeenCalledWith(400);
+    expect(res.json).toHaveBeenCalledWith({ message: 'Invalid category' });
+  });
+
+  it('should handle errors', async () => {
+    const req = { params: { category: 'Travel' } };
```

Comment:
If we cannot refactor, kindly mention the specification of what type of error needs to be handled as done in the above test cases